flutter/lib/ui/settings/task_config_screen.dart

```diff
@@ -47,25 +196,57 @@ class TaskConfigScreen extends StatelessWidget {
     );
   }
 
+  Widget _makeCacheFolderNotice(AppLocalizations l10n) {
+    if (defaultCacheFolder.isEmpty) {
+      return const SizedBox.shrink();
+    }
+    return Column(
+      children: [
+        ListTile(
+          title: Padding(
+            padding: const EdgeInsets.only(bottom: 5),
+            child: Text(
+              l10n.settingsTaskCacheFolderTitle,
+            ),
+          ),
+          subtitle: Text(l10n.settingsTaskCacheFolderDesc),
+        ),
+        ListTile(
+          enabled: false,
+          title: Text(l10n.settingsTaskCacheFolderDefault),
+          subtitle: const Text(defaultCacheFolder),
+          leading: const Radio<bool>(
+            value: true,
+            groupValue: true,
+            onChanged: null,
+          ),
+        )
+      ],
+    );
+  }
+
   @override
   Widget build(BuildContext context) {
-    final stringResources = AppLocalizations.of(context);
+    final l10n = AppLocalizations.of(context);
     final store = context.watch<Store>();
 
     return Scaffold(
       appBar: AppBar(
-        title: Text(stringResources.settingsTaskConfigTitle),
+        title: Text(l10n.settingsTaskConfigTitle),
       ),
-      body: ListView.builder(
-        itemCount: _configs.length,
-        itemBuilder: (context, index) {
-          final configuration = _configs[index];
-          return getOptionPattern(
-            context,
-            configuration,
-            store.chosenConfigurationName,
-          );
-        },
+      body: ListView(
+        children: [
+          if (!Platform.isIOS) ...[
+            _makeCacheFolderNotice(l10n),
+            _DataFolderSelectorHelper(context).build(),
+            const Divider(),
+          ],
```

Comment:
Because iOS prevents access to folders outside of the app folder.
If you set it on iOS, the app will crash during benchmark because native code is unable to read files.
When I tried reading files in Flutter, I get this exception:
```
FileSystemException (FileSystemException: Directory listing failed, path = ‘/private/var/mobile/Containers/Shared/AppGroup/E817F925-A312-4F1F-A333-123A9849B601/File Provider Storage/Downloads/’ (OS Error: Operation not permitted, errno = 1))
```

I additionally checked how custom file managers handle this, and it seems like they don't have such problems. However, I wasn't able to find an instruction for this.

I also think we may get troubles during review from Apple when we publish the app, so I don't think it worth spending time solving this.

Reply:
In iOS, it's possible and legit to access files outside the application directory, the app just needs to ask for it.
The `/private` prefix means that that path is protected, and the app needs to ask for permission to access it. In iOS, that can be done with `CFURLStartAccessingSecurityScopedResource` or `startAccessingSecurityScopedResource`. I did a quick search in the source code of the `file_picker` plugin and did not find that methods so probably they don't request that permission. It would take quite an effort to solve this so I think we can just ignore it for now, and do it later if there is a request for this feature.

Reply:
Please add a comment so we later still know why iOS is not supported.

Reply:
Done